dra-evolution/testdata/classes.yaml

```diff
@@ -1,162 +1,208 @@
+---
 # Request an example.com PF
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: example.com-sriov-pf
-spec:
-  driver: example.com-sriov
-  constraints: "device.sriovType == 'pf'"
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.stringAttributes['type.nic.k8s.io'] == 'pf'"
 ---
 # Request an example.com VF
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: example.com-sriov-vf
-spec:
-  driver: example.com-sriov
-  constraints: "device.sriovType == 'vf'"
----
-#######
-# NOTE: We cannot create a class for example.org PF or VF, because instead of
-# using attributes, they use custom vf/pf resource types to control this. Claims
-# can request one or the other by using `requests`.
-#######
-#
-# ConfigMap to configure a DRA driver that manages VLAN interfaces.
-# NOTE: How do we control permissions on what these configs can do?
-apiVersion: v1
-kind: ConfigMap
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.stringAttributes['type.nic.k8s.io'] == 'vf'"
+---
+# Request an example.com PF or VF
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
-  namespace: default
-  name: vlan-driver-vlan-2000
-data:
-  vlanID: 2000
+  name: example.com-sriov-vf
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.stringAttributes['type.nic.k8s.io'] == 'vf' || device.stringAttributes['type.nic.k8s.io'] == 'pf'"
 ---
 # Request a VLAN interface on VLAN 2000
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: vlan-2000
-spec:
-  deviceType: vlan
-  configs:
-  - apiVersion: v1
-    kind: ConfigMap
-    namespace: default
-    name: vlan-driver-vlan-2000
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.stringAttributes['type.k8s.io'] == 'vlan'"
+claim:
+  config:
+  - vendor:
+      driverName: example.com-sriov
+      parameters:
+        ## ConfigMap to configure an example.com-sriov driver that manages VLAN interfaces.
+        ## NOTE: How do we control permissions on what these configs can do?
+        apiVersion: v1
+        kind: ConfigMap
+        data:
+          vlanID: 2000
+    # The fictional NIC KEP could also add an in-tree type for "NIC configuration",   which
+    # then would replace the "vendor" entry above:
+    # nic:
+    #   vlanID: 2000
+---
+# Request a Foozer GPU and some NIC.
+#
+# The other classes implicitly have
+#   defaultRequests: [ {} ]
+#
+# as default and thus ask for one device that matches the class criteria.
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
+metadata:
+  name: example.com-foozer-and-nic
+defaultRequests:
```

Comment:
I don't understand this. Why are there two devices in the class? Is the class being used as a sort of pre-defined constellation of devices?

Reply:
Yes, that was the intent. It's on the chopping block of bad ideas, I'm just waiting for confirmation to remove it again.